R/utils.R

```diff
@@ -220,6 +219,16 @@ read_from_url <- function(resource, ...){
   # where that's not the case
   message("Reading the data using the ", fun$fun, " function from the ",
           fun$package, " package.")
+
+  if (reported_format %in% c("xls", "xlsx")) {
+    sheets <- readxl::excel_sheets(tmp)
+    if (length(sheets) > 1 && !"sheet" %in% names(list(...))) {
```

Comment:
I think this can be:

```suggestion
    if (length(sheets) > 1 && !hasArg("sheet")) {
```

Reply:
Also, this function is getting a little unwieldy - do you think there any parts that make sense to abstract out?

Reply:
`handle_excel`?

https://github.com/bcgov/bcdata/blob/068716891ecdde317b19e5177a2d39c75d4293f4/R/utils.R#L223-L230

Reply:
I like it!

Reply:
What happens if `sheet` passed by the user doesn't match a sheet in the file? Does `readxl` give a good error message or do we need to catch that?